### ISSUE TITLE
AIP-51 - Executor Specific CLI Commands

### ISSUE THREAD

**Issue by o-nikolas** [apache/airflow#27932]
### Overview
Some Executors have their own first class CLI commands (now that’s hardcoding/coupling!) which setup or modify various components related to that Executor.
### Examples
- **5a**) Celery Executor commands: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L1689-L1734
- **5b**) Kubernetes Executor commands: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L1754-L1771
- **5c**) Default CLI parser has hardcoded logic for Celery and Kubernetes Executors specifically: https://github.com/apache/airflow/blob/27e2101f6ee5567b2843cbccf1dca0b0e7c96186/airflow/cli/cli_parser.py#L63-L99
### Proposal
Update the BaseExecutor interface with a pluggable mechanism to vend CLI `GroupCommands` and parsers. Executor subclasses would then implement these methods, if applicable, which would then be called to fetch commands and parsers from within Airflow Core cli parser code. We would then migrate the existing Executor CLI code from cli_parser to the respective Executor class.

Pseudo-code example for vending `GroupCommand`s:
```python
# Existing code in cli_parser.py
...
airflow_commands: List[CLICommand] = [
GroupCommand(
        name='dags',
        help='Manage DAGs',
        subcommands=DAGS_COMMANDS,
    ),
    ...
]
# New code to add groups vended by executor classes
executor_cls, _ = ExecutorLoader.import_executor_cls(conf.get('core', 'EXECUTOR'))
airflow_commands.append(executor_cls.get_cli_group_commands())
...
```

**Comment by potiuk** [comment]
Hey @o-nikolas (cc: @eladkal). I realized there is one more thing to do after https://github.com/apache/airflow/pull/29055 : making sure our documentation describes the CLI commands in each executor.

Currently in https://github.com/apache/airflow/blob/main/docs/apache-airflow/cli-and-env-variables-ref.rst we are using sphinx-argparse to generate list of commands, but once we moved the celery and kubernetes commands to the executors, we also need to move documentation for those commands there as well - in a very similar way I've moved documentation for configuration https://github.com/apache/airflow/pull/32604/files#diff-b9c256e60460840fb031abc0daa8cb160175593ac03b358c426fc35f9ed4f3f0 

This is a documentation only, so this is not a blocker to release providers. We still have time to get it implemented in main for 2.7.0 and we can later apply it on top of the provider TAG and rebuild/upload the documentation for celery and kubernetes. How I imagine it, there should be a section where we list all the providers that have executors with some commands added, and have links to the new "CLI" pages where those providers display the args (also done via sphinx-argparse - but only using the commands returned by the executors. I think we have all that we need for that, the provider.yaml files already have optional "executors" dictionary - listing all executors that can contribute the commands (some de-duplication will be needed because for example Celery and CeleryKubernetes contribute the same commands). We could instantiate those args and pass them to sphinx-argparse to generate the documentation.

Happy to help @o-nikolas if needed.
